api/src/main/java/redux/Store.java

```diff
@@ -0,0 +1,95 @@
+package redux;
+
+/**
+ * Coordinates actions and [Reducers][Reducer]. Store has the following responsibilities:
+ * <ul>
+ * 	<li>Holds application state</li>
+ * 	<li>Allows access to state via {@link #getState()}</li>
+ * 	<li>Allows state to be updated via {@link redux.Dispatcher#dispatch(Object)}</li>
+ * 	<li>Registers listeners via {@link #subscribe(Subscriber)}</li>
+ * 	<li>Handles unregistering of listeners via the {@link Subscriber} returned by {@link #subscribe(Subscriber)}</li>
+ * </ul>
+ *
+ * @see <a href="http://redux.js.org/docs/basics/Store.html">http://redux.js.org/docs/basics/Store.html</a>
+ */
+public interface Store<S> extends Dispatcher {
+
+	/**
+	 * Returns the current state tree of your application. It is equal to the last value returned by the store’s
+	 * reducer.
+	 *
+	 * @return the current state
+	 * @see <a href="http://redux.js.org/docs/api/Store.html#getState">http://redux.js.org/docs/api/Store.html#getState</a>
+	 */
+	S getState();
+
+	/**
+	 * Adds a change listener. It will be called any time an action is dispatched, and some part of the state tree may
+	 * potentially have changed. You may then call {@link #getState()} to read the current state tree inside the callback.
+	 *
+	 * @return A subscription
+	 * @param subscriber The subscriber
+	 * @see <a href="http://redux.js.org/docs/api/Store.html#subscribe">http://redux.js.org/docs/api/Store.html#subscribe</a>
+	 */
+	Subscription subscribe(Subscriber subscriber);
+
+	/**
+	 * Replaces the reducer currently used by the store to calculate the state.
+	 *
+	 * @param reducer The reducer
+	 * @see <a href="http://redux.js.org/docs/api/Store.html#replaceReducer">http://redux.js.org/docs/api/Store.html#replaceReducer</a>
+	 */
+	void replaceReducer(Reducer<S> reducer);
+
+	/**
+	 * An interface that creates a Redux store.
+	 *
+	 * @see <a href="http://redux.js.org/docs/Glossary.html#store-creator">http://redux.js.org/docs/Glossary.html#store-creator</a>
+	 */
+	interface Creator<S> {
```

Comment:
I am curious why do you think it is better to add the type to the creator ? 

To my understanding, a store creator knows how to create a store given a reducer and a state. I don't think it needs to carry a type.

I also don't think it is worth it because : 
- it is not the vanilla API (event if it is implicit) 
- we lose type safety (otherwise it [works fine](https://github.com/glung/jvm-redux-devtools-instrument/blob/master/lib/src/main/kotlin/redux/devtools/DevTools.kt#L71))
- It would work for your store and all stores out there (cf adapters here https://github.com/glung/jvm-redux)